File: chessgame-api/src/main/java/git/dimitrikvirik/chessgameapi/model/domain/UserAccount.java
//package git.dimitrikvirik.chessgameapi.model.domain;
//
//
//import lombok.Data;
//
//import javax.persistence.*;
//import java.util.List;
//
//@Entity
//@Data
//public class UserAccount {
//    @Id
//    @GeneratedValue(
//            strategy = GenerationType.IDENTITY
//    )
//    private Long id;
//    private String username;
//    private String email;
//    private String keycloakId;
//    @ManyToMany
//    private List<UserAccount> friends;
//    @OneToMany(mappedBy = "whitePlayer")
//    private List<Game> whitePlayerGames;
//    @OneToMany(mappedBy = "blackPlayer")
//
//    private List<Game> blackPlayerGames;
//    @OneToMany(mappedBy = "winnerPlayer")
//
//    private List<Game> wonGames;
//}

File: chessgame-api/src/main/kotlin/git/dimitrikvirik/chessgameapi/model/domain/UserAccount.kt
package git.dimitrikvirik.chessgameapi.model.domain

import javax.persistence.*

@Entity
@Table(name = "user_account")
class UserAccount(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    val username: String? = null,
    @ManyToMany
    val friends: List<UserAccount>? = null,
    @OneToMany(mappedBy = "whitePlayer")
    val whitePlayerGames: List<Game>? = null,
    @OneToMany(mappedBy = "blackPlayer")
    val blackPlayerGames: List<Game>? = null,
    @OneToMany(mappedBy = "winnerPlayer")
    val wonGames: List<Game>? = null
)